draft-ietf-sipcore-sip-push.xml

```diff
@@ -50,78 +50,90 @@
+        associated with the push notification registration. The UA will use a REGISTER request 
+        to provide the PRID to the SIP proxy, which will then request that push notifications 
+        are sent to the UA.
       </t>
+      <!-- ajm: I heavily modified this paragraph below to match what Figure 1 shows. Please check
+           for accuracy. -->
       <t>
-        When the proxy receives  
-        a SIP request for a new dialog or a stand-alone SIP request addressed towards a UA, 
-        or when the proxy determines that the UA needs to send a binding-refresh REGISTER request, the proxy will 
-        request that a push notification is sent to the UA, using the PNS of the UA. Once the UA receives 
-        the push notification, it will be able to send a binding-refresh REGISTER request and receive 
-        the incoming SIP request. The proxy will receive the REGISTER request. If the push notification request was
-        triggered by a SIP request addressed towards the UA (see above), once the REGISTER request has been 
-        accepted by the SIP registrar <xref target="RFC3261"/>, and the associated SIP 2xx response has been forwarded by the proxy towards the UA, 
-        the proxy can forward the SIP request towards the UA using normal SIP routing procedures. In some cases the proxy can forward the SIP
-        request without waiting for the SIP 2xx response to the REGISTER request. Note that this mechanism necessarily adds
-        delay to responding to requests requiring push notification. The consequences of that delay are discussed
-        in <xref target="section.proxy.req.oth"/>.
+        When the SIP proxy receives a SIP request for a new dialog or a stand-alone SIP request 
+        addressed towards a UA, or when the SIP proxy determines that the UA needs to send a 
+        binding-refresh REGISTER request, the SIP proxy will send a push request containing the 
+        PRID of the UA to the PNS, which will then send a push notification to the UA. Once the 
+        UA receives the push notification, it will be able to send a binding-refresh REGISTER 
+        request. The proxy receives the REGISTER request from the UA and forwards it to the SIP 
+        registrar <xref target="RFC3261"/>. After accepting the REGISTER request, the SIP registrar 
+        sends a 2xx response to the proxy, which forwards the response to the UA. If the push 
+        notification request was triggered by a SIP request addressed towards the UA, the proxy 
+        can then forward the SIP request to the UA using normal SIP routing procedures. In some 
+        cases the proxy can forward the SIP request without waiting for the SIP 2xx response to 
+        the REGISTER request from the SIP registrar. Note that this mechanism necessarily adds 
+        delay to responding to requests requiring push notification. The consequences of that 
+        delay are discussed in <xref target="section.proxy.req.oth"/>.
       </t>
```

Comment:
The text talks about the SIP proxy sending a push request to the PNS. Elsewhere in the document the text talks about the SIP proxy requesting that a push notification is sent to the UA. That wording is based on IESG comments by Ekr, so I would like to keep it.

Reply:
Did Ekr send his feedback on the wording to the sipcore list? I can't find this particular suggestion in his "Discuss" emails (other than he didn't like the construction of "request ... towards".

Reply:
I don't know when/if he sent the feedback, but at least he didn't object to the wording.

Reply:
The wording used throughout the normative text ("request that a push notification is sent", 18 instances) is passive (the PNS is left out), but I didn't touch it. I understand that the draft is focusing on SIP specifics. However, I felt that the introductory text could provide more details on the call flow given in Figure 1 and help the reader orient themselves in the PNS network.

Reply:
Ack. If you want, you could place the reference to Figure 1 (<xref target="fig-sip-pn-arch"/> shows the generic push notification architecture supported by the mechanism in this document.) in front of "When a UA registers with a PNS..." to tie the text and figure together.

Reply:
I can do that.